connectors/sync_job_runner.py

```diff
@@ -112,7 +114,9 @@ async def execute(self):
                 options=bulk_options,
             )
 
+            self._reporting_cycle_start_time = time.time()
```

Comment:
Id' probably move it to `check_job`. Actually it might be even nicer to encapsulate this logic into a class that can be reusable in other classes. We have similar logic (do something every now and then) in other places - heartbeat, job cleanup, this.